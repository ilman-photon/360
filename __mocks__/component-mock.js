jest.mock("@fontsource/roboto", () => {
  return jest.fn();
});
jest.mock("@fontsource/libre-franklin", () => {
  return jest.fn();
});
jest.mock("next/dynamic", () => ({
  __esModule: true,
  default: (...props) => {
    const matchedPath = /(.)*(\'(.*)\')(.)*/.exec(props[0].toString());
    if (matchedPath) return require(matchedPath[3]);
    // eslint-disable-next-line react/display-name
    else return () => <></>;
  },
}));

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
  withRouter: (component) => {
    component.defaultProps = {
      ...component.defaultProps,
      router: { pathname: "mocked-path" },
    };
    return component;
  },
}));

jest.mock("next-i18next", () => ({
  use: () => {
    return {
      init: () => {},
    };
  },
  t: (k) => k,
  appWithTranslation: (Component) => {
    return Component;
  },
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
  withRouter() {
    return jest.fn();
  },
}));

jest.mock("react-idle-timer", () => ({
  useIdleTimer: jest.fn().mockReturnValue({
    getRemainingTime: jest.fn(),
    isPrompted: jest.fn(),
    activate: jest.fn(),
  }),
}));
