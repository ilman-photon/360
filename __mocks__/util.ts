
/**
 * Sleeps for x amount of milliseconds.
 *
 * @param time   Amount of time in milliseconds.
 * @returns Promise resolving timeout id.
 * @private
 */
export function sleep (time: number = 0): Promise<number> {
  return new Promise(resolve => setTimeout(resolve, time))
}

interface IWaitForOptions {
  timeout?: number
  interval?: number
}

/**
 * Waits for the given function to return true.
 *
 * @param fn Function to check
 * @param opts Options
 * @param opts.timeout Timeout before test fails
 * @param opts.interval Interval to check function
 * @return {Promise}
 */
export async function waitFor (fn: (...args: any[]) => boolean, options: IWaitForOptions = { timeout: 1000, interval: 50 }): Promise<void> {
  let elapsed = 0

  const run = async () => {
    if (elapsed >= options.timeout) {
      throw new Error(`‚ùå waitFor reached timeout of ${options.timeout}ms`)
    }

    if (fn()) return
    await sleep(options.interval)
    elapsed += options.interval
    await run()
  }

  await run()
}