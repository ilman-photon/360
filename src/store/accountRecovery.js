import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { Api } from "../pages/api/api";

export const fetchPatientAccount = createAsyncThunk(
  "accountRecovery/fetchPatientAccount",
  async ({ keyword }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/getPatientDetails/${keyword}`;
    return api.getResponse(url, null, "get");
  }
);

export const onActivate = createAsyncThunk(
  "accountRecovery/onActivate",
  async ({ patientId }) => {
    const url = `/api/dummy/account-recovery/onActivate/${patientId}`;
    return fetch(url, { method: "POST" }).then((res) => res.json());
  }
);

export const onUnlock = createAsyncThunk(
  "accountRecovery/onUnlock",
  async ({ patientId }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/unlockAccountByAdmin/${patientId}`;

    try {
      const response = await api.getResponse(url, null, "get");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onSendPasswordReset = createAsyncThunk(
  "accountRecovery/onSendPasswordReset",
  async ({ patientId, patientData, selectedCommunication }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/sendPasswordResetLinkToUser`;
    const domain = window.location.origin;
    const NEXT_PUBLIC_ONE_TIME_LINK = process.env.NEXT_PUBLIC_ONE_TIME_LINK;
    const postBody = {
      patient: {
        userName: patientData.username,
      },
      preferredComunication: selectedCommunication,
      resetPassword: true,
      link: `${domain}${NEXT_PUBLIC_ONE_TIME_LINK}`,
    };

    try {
      const response = await api.getResponse(url, postBody, "post");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onShareUsername = createAsyncThunk(
  "accountRecovery/onShareUsername",
  async ({ patientId, selectedCommunication }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/shareUserNameByAdmin`;
    const postBody = {
      patient: {
        id: patientId,
      },
      preferredComunication: selectedCommunication,
    };

    try {
      const response = await api.getResponse(url, postBody, "post");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onViewSecurityQuestions = createAsyncThunk(
  "accountRecovery/onViewSecurityQuestions",
  async ({ patientId }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/viewSecurityQuestions/${patientId}`; // d3724cd1-ebae-4f1a-82a5-544ff33b0313

    try {
      const response = await api.getResponse(url, null, "get", false);

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

const getPatientStatus = (data) => {
  if (data.verified != "active") return "inactive";
  switch (data.status) {
    case "locked":
    case "L":
      return "locked";
    case "unlocked":
    default:
      return "unlocked";
  }
};

const buildAccountData = (data) => {
  return data.map((item) => {
    return {
      ...item,
      name: item.patientName || item.name,
      email: item.emailId,
      phone: item.phoneNumber,
      status: getPatientStatus(item),
    };
  });
};

const INITIAL_STATE = {
  patientList: [],
  securityQuestions: [],
  securityQuestionsRaw: {},
  status: null,
  error: null,
};

export const accountRecoveryStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setStatus: (state, { payload }) => {
      state.status = payload;
    },
    setAccountDataById: (state, { payload }) => {
      const updatedState = state.patientList.map((item, idx) => {
        const id = item.id || item.patientId;
        const payloadId = payload.id || payload.patientId;
        if (payloadId === id) {
          item = payload;
        }
        return item;
      });

      state.patientList = buildAccountData(updatedState);
    },
  },
  extraReducers: {
    [fetchPatientAccount.pending]: (state) => {
      state.status = "loading";
    },
    [fetchPatientAccount.fulfilled]: (state, { payload }) => {
      state.patientList = buildAccountData(payload);
      state.status = "success";
    },
    [fetchPatientAccount.rejected]: (state, action) => {
      state.error = action;
      state.status = "failed";
    },
    [onViewSecurityQuestions.fulfilled]: (state, { payload }) => {
      const response = payload.response;
      let mapped = [];
      if (response?.ResponseCode === 3000) {
        const securityQuestions = response.SecurityQuestions[0];
        mapped = Object.keys(securityQuestions).map((keys, item) => {
          return {
            question: keys,
            answer: securityQuestions[keys],
          };
        });

        state.securityQuestionsRaw = response.SecurityQuestions;
      }

      state.securityQuestions = mapped;
      state.status = "success";
    },
    [fetchPatientAccount.rejected]: (state, action) => {
      state.error = action;
      state.securityQuestions = [];
    },
  },
});

// Action creators are generated for each case reducer function
export const { setAccountDataById, setStatus } = accountRecoveryStore.actions;

export default accountRecoveryStore.reducer;
