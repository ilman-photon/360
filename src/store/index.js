import { createSlice } from "@reduxjs/toolkit";

const INITIAL_STATE = {
  loading: true,
  counter: 0,
};

export const indexStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setLoading: (state, action) => {
      state.loading = action.payload;
    },
    increment: (state, action) => {
      state.counter++;
    },
  },
});

// Action creators are generated for each case reducer function
export const { setLoading, increment } = indexStore.actions;

export default indexStore.reducer;
