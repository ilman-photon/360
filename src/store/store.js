import { configureStore } from "@reduxjs/toolkit";
import index from "../store/index";
import user from "../store/user";

export const store = configureStore({
  reducer: {
    user,
    index,
  },
});

export default store;
