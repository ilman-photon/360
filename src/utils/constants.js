module.exports = {
  EMPTY_STRING: "",
  H1: "h1",
  H2: "h2",
  SMALL: "small",
  LARGE: "large",
  PATIENT: "patient",
  PRIMARY: "primary",
  SECONDARY: "secondary",
  MIDDLE: "middle",
  CENTER: "center",
  FLEX_END: "flex-end",
  BODY_REGULAR: "bodyRegular",
  FILLED: "filled",
  INPUT_TEXT: "text",
  INPUT_PASSWORD: "password",
  ACCOUNT_LOCK_COUNT: 3,
  SUBMIT: "submit",
  SELECT_OPTION: "selectOption",
  SECURITY_QUESTION: "securityQuestion",
  ONE_TIME_LINK: "oneTimeLink",
  PASSWORD_RESET: "passwordReset",
  BOTH: "both",
  MODE_COMMUNICATION_KEY: "modeComunication",
  EMAIL: "email",
  PHONE: "phone",
  YES: "yes",
  MFA_COMPONENT_NAME: "MultiFactorAuthentication",
  SQ_COMPONENT_NAME: "SecurityQuestion",
  MONTH_NAME: [
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec",
  ],
  DAY_NAME: ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
  TEST_ID: {
    LOGIN_TEST_ID: {
      email: "emailorphonenumber",
      password: "password",
      forgotLink: "forgotpswd",
      loginBtn: "loginbtn",
      guestBtn: "continueasguest",
      createAccountBtn: "createaccount",
      syncAppointmentLink: "syncAppointmentLink",
    },
    FORGOT_TEST_ID: {
      email: "emailorphonenumber",
      continueBtn: "continuebtn",
      loginLink: "backtologin",
      answerQuestions: "answerquestions",
      oneTimeLink: "onetimelink",
      securityQuestion: "securityquestion",
    },
    REGISTER_TEST_ID: {
      firstname: "firstname",
      lastname: "lastname",
      dateofbirth: "dateofbirth",
      calander: "calander",
      email: "email",
      mobilenumber: "mobilenumber",
      password: "password",
      emailradio: "emailradio",
      phoneradio: "phoneradio",
      bothradio: "bothradio",
      registerbtn: "registerBtn",
      loginlink: "loginlink",
    },
    SCHEDULE_GUEST_TEST_ID: {
      firstname: "firstname",
      lastname: "lastname",
      dateofbirth: "dateofbirth",
      calander: "calander",
      email: "email",
      mobilenumber: "mobilenumber",
      password: "password",
      emailradio: "emailradio",
      phoneradio: "phoneradio",
      bothradio: "bothradio",
      registerbtn: "registerBtn",
      signInlink: "signInlink",
    },
    HOME_TEST_ID: {
      appoinments: "appoinments",
      myhealthchart: "myhealthchart",
      mycareteam: "mycareteam",
      labresults: "labresults",
      billing: "billing",
      username: "username",
      account: "account",
      logout: "logout",
    },
    MFA_TEST_ID: {
      radioEmail: "loc_RadioEmail",
      radioPhone: "loc_RadioPhone",
      rememberMe: "loc_chkRemember",
      btnConfirm: "loc_btnConfirm",
      btnBack: "loc_btnBack",
      inputCode: "loc_txtEnterCode",
      btnSubmit: "loc_btnSubmit",
      btnResend: "loc_btnResend",
      btnSubmitSecurity: "loc_btnSubmitSecurity",
      btnSkip: "loc_btnSkip",
    },
    PERSONAL_INFO_TEST_ID: {
      txtName: "loc_txtName",
      btnSave: "loc_btnSave",
      btnCancel: "loc_btnCancel",
      btnDelete: "loc_btnDelete",
    },
    APPOINTMENT_TEST_ID: {
      location: "location",
      searchbtn: "searchbtn",
      filterbtn: "filterbtn",
      locationInput: "locationInput",
      dateInput: "dateInput",
      purposeInput: "purposeInput",
      insuranceInput: "insuranceInput",
      FILTER_RESULT: {
        container: "appointment_filter_result_container",
      },
      FILTER_COMPONENT: {
        drawer: "appointment_filter_component_drawer",
      },
    },
    BIOGRAPHY_TEST_ID: {
      about: "biography-about",
      location: "biography-location",
      insurance: "biography-insurance",
      education: "biography-education",
      viewAll: "biography-view-all-insurance",
    },
    INSURANCE_TEST_ID: {
      addButton: "loc_btnAddInsurance",
      provider: "loc_inputProvider",
      planName: "loc_inputPlan",
      subscriberId: "loc_inputSubsId",
      group: "loc_inputGroup",
      save: "loc_btnSaveInsuranve",
      cancel: "loc_btnCancelInsuranve",
      uploadFrontImage: "loc_btnUploadFront",
      uploadBackImage: "loc_btnUploadBack",
    },
    SEARCH_PROVIDER_TEST_ID: {
      viewAll: "loc_viewAll",
    },
    SUBNAVIGATION: "subNavigation-link",
    SCHEDULE_APPOINTMENT_TEST_ID: {
      APPOINTMENT_LOCATION: {
        editButton: "schedule_appointment_location_edit_button",
      },
      APPOINTMENT_DETAILS: {
        editButton: "schedule_appointment_details_edit_button",
      },
      step2Button: "schedule_appointment_step2_button",
    },
  },
};
