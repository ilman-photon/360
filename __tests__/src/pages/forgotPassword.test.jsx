import { render, waitFor, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import store from "../../../src/store/store";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";
import ForgotPasswordPage from "../../../src/pages/patient/forgot-password";
import constants from "../../../src/utils/constants";
import { renderForgotPassword } from "../../../__mocks__/commonSteps";
import { useRouter } from "next/router";

describe("Forgot Password", () => {
  let container;
  const mock = new MockAdapter(axios);
  const { FORGOT_TEST_ID } = constants.TEST_ID;
  beforeEach(() => {
    container = render(
      <Provider store={store}>
        {ForgotPasswordPage.getLayout(<ForgotPasswordPage />)}
      </Provider>
    );
  });
  test("renders forgot password", () => {
    expect(container.getByTestId(FORGOT_TEST_ID.loginLink)).toBeInTheDocument();
    const button = container.getByTestId(FORGOT_TEST_ID.loginLink);
    fireEvent.click(button);
  });

  const emailFieldValidation = () => {
    const usernameField = container.getByLabelText(/usernamePlaceHolder/i);
    fireEvent.change(usernameField, { target: { value: "user1@gmail.com" } });
    expect(usernameField.value).toEqual("user1@gmail.com");
  };
  test("check valid email", emailFieldValidation);

  const resetPassword = async () => {
    emailFieldValidation();
    const expectedResult = {
      ResponseCode: 1000,
      ResponseType: "success",
      SecurityQuestions: [
        {
          "Where did you go the first time you flew on a plane?": "America",
          "Who is your all-time favorite movie character": "Peter",
          "In what city or town did your parents meet?": "Berlin",
        },
      ],

      PreferredComunication: "Both",
    };
    mock.onPost(`/ecp/patient/validate`).reply(200, expectedResult);
    const button = container.getByTestId(FORGOT_TEST_ID.continueBtn);
    fireEvent.click(button);
    await waitFor(() => container.getByText("or"));
    expect(container.getByText("or")).toBeInTheDocument();
  };
  test("click reset password", resetPassword);

  test("click answer onetimelink", async () => {
    emailFieldValidation();
    const expectedResultValidate = {
      ResponseCode: 1000,
      ResponseType: "success",
      SecurityQuestions: [],

      PreferredComunication: "Both",
    };
    mock.onPost(`/ecp/patient/validate`).reply(200, expectedResultValidate);
    const continueBtn = container.getByTestId(FORGOT_TEST_ID.continueBtn);
    fireEvent.click(continueBtn);
    await waitFor(() => container.getByText("or"));
    mock.reset();
    const expectedResult = {
      ResponseCode: 1000,
      ResponseType: "success",
    };
    mock.onPost(`/ecp/patient/onetimelink`).reply(200, expectedResult);
    const button = container.getByTestId(FORGOT_TEST_ID.oneTimeLink);
    fireEvent.click(button);
    await waitFor(() => container.getByText(`Mode of Communication`));
    fireEvent.click(container.getByDisplayValue("email"));
    fireEvent.click(container.getByTestId(FORGOT_TEST_ID.oneTimeLink));

    await waitFor(() => container.getByText(`successLabel`));
  });

  const selectOptions = async () => {
    await resetPassword();
    const button = container.getByTestId(FORGOT_TEST_ID.answerQuestions);
    fireEvent.click(button);
    await waitFor(() =>
      container.getByText(
        `Where did you go the first time you flew on a plane?`
      )
    );
  };
  test("click answer questions", async () => {
    await selectOptions();
  });

  test("click answer questions > reset password", async () => {
    // emailFieldValidation();
    // const expectedResultValidate = {
    //   ResponseCode: 1000,
    //   ResponseType: "success",
    //   SecurityQuestions: [],

    //   PreferredComunication: "none",
    // };
    // mock.onPost(`/ecp/patient/validate`).reply(200, expectedResultValidate);
    // const continueBtn = container.getByTestId(FORGOT_TEST_ID.continueBtn);
    // fireEvent.click(continueBtn);
    // await waitFor(() => container.getByText("or"));
    // mock.reset();
    // const expectedResult = {
    //   ResponseCode: 1000,
    //   ResponseType: "success",
    //   token: 1182,
    // };
    // mock.onPost(`/ecp/patient/onetimelink`).reply(200, expectedResult);
    // const button = container.getByTestId(FORGOT_TEST_ID.oneTimeLink);
    // fireEvent.click(button);
    // await waitFor(() => container.getByTestId(FORGOT_TEST_ID.continueBtn));
    // fireEvent.click(container.getByTestId(FORGOT_TEST_ID.continueBtn));
    // await waitFor(() => container.getByText("successSentLinkTitleOneTime"));
    // fireEvent.click(container.getByTestId(FORGOT_TEST_ID.continueBtn));
    // await waitFor(() => container.getByText(`successLabel`));
    // const expectedResultReset = {
    //   ResponseCode: 1000,
    //   ResponseType: "success",
    //   email: "smith1@photon.com",
    // };
    // mock
    //   .onPost(`/ecp/patient/resetPasswordLink`)
    //   .reply(200, expectedResultReset);
    // fireEvent.click(container.getByTestId(FORGOT_TEST_ID.oneTimeLink));
  });

  test("click answer questions back to login", async () => {
    await selectOptions();
    const button = container.getByTestId(FORGOT_TEST_ID.loginLink);
    fireEvent.click(button);
  });

  test("answer questions continue", async () => {
    selectOptions();
    const Q1 = container.getByLabelText(/usernamePlaceHolder/i);
    fireEvent.change(Q1, { target: { value: "user1@gmail.com" } });
  });

  test("Sync Email Appointment failed validate", async () => {
    const useRouter = jest.spyOn(require("next/router"), "useRouter");
    useRouter.mockReturnValue({
      back: jest.fn(),
      asPath: "/patient/sync",
      push: jest.fn(),
    });
    const usernameField = container.getByLabelText(/usernamePlaceHolder/i);
    fireEvent.change(usernameField, {
      target: { value: "patient1@gmail.com" },
    });
    expect(usernameField.value).toEqual("patient1@gmail.com");
    const expectedResult = {
      ResponseCode: 1001,
      ResponseType: "failure",
    };
    mock.onPost(`/ecp/patient/validate`).reply(404, expectedResult);

    expect(container.getByTestId("continuebtn")).toBeInTheDocument();
    fireEvent.click(container.getByTestId("continuebtn"));
  });
});
