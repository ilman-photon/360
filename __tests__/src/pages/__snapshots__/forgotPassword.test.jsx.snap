// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password renders forgot password 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="authLayout"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1nft05v-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27124%27%20height=%2736%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Eyecare Provider name with logo"
                  aria-label="Eyecare Provider name with logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </header>
        <div
          class="authContainer"
        >
          <div
            class="authComponentContainer MuiBox-root css-1h4hxqr"
          >
            <div
              class="containerPage MuiBox-root css-0"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-1pjr2pz-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                  style="display: flex; flex-direction: column; padding: 0px;"
                >
                  <h2
                    aria-label="title heading"
                    aria-roledescription=""
                    class="MuiTypography-root MuiTypography-h2 css-izw0or-MuiTypography-root"
                  >
                    title
                  </h2>
                  <form
                    style="display: grid; margin-top: 8px;"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root custom-input css-12caibx-MuiFormControl-root-MuiTextField-root"
                      data-testid="emailorphonenumber"
                      withicon="true"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-1suhfmo-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="false"
                        for="username"
                        id="username-label"
                      >
                        usernamePlaceHolder
                      </label>
                      <div
                        class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-jq7ton-MuiInputBase-root-MuiFilledInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
                          id="username"
                          placeholder=""
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <button
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonSmall  css-mn6rr9-MuiButtonBase-root-MuiButton-root"
                      data-testid="continuebtn"
                      style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center;"
                      tabindex="0"
                      type="submit"
                    >
                      resetPasswordText
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </form>
                  <a
                    aria-label="backButtonLink link"
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-160akwi-MuiTypography-root-MuiLink-root"
                    data-testid="backtologin"
                    style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center; cursor: pointer;"
                  >
                    backButtonLink
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg authImageContainer css-r7w29g-MuiContainer-root"
          >
            <div
              class="imageBannerContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt=""
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="authLayout"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1nft05v-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27124%27%20height=%2736%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Eyecare Provider name with logo"
                aria-label="Eyecare Provider name with logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </header>
      <div
        class="authContainer"
      >
        <div
          class="authComponentContainer MuiBox-root css-1h4hxqr"
        >
          <div
            class="containerPage MuiBox-root css-0"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-1pjr2pz-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                style="display: flex; flex-direction: column; padding: 0px;"
              >
                <h2
                  aria-label="title heading"
                  aria-roledescription=""
                  class="MuiTypography-root MuiTypography-h2 css-izw0or-MuiTypography-root"
                >
                  title
                </h2>
                <form
                  style="display: grid; margin-top: 8px;"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root custom-input css-12caibx-MuiFormControl-root-MuiTextField-root"
                    data-testid="emailorphonenumber"
                    withicon="true"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-1suhfmo-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="username"
                      id="username-label"
                    >
                      usernamePlaceHolder
                    </label>
                    <div
                      class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-jq7ton-MuiInputBase-root-MuiFilledInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
                        id="username"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonSmall  css-mn6rr9-MuiButtonBase-root-MuiButton-root"
                    data-testid="continuebtn"
                    style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center;"
                    tabindex="0"
                    type="submit"
                  >
                    resetPasswordText
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
                <a
                  aria-label="backButtonLink link"
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-160akwi-MuiTypography-root-MuiLink-root"
                  data-testid="backtologin"
                  style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center; cursor: pointer;"
                >
                  backButtonLink
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg authImageContainer css-r7w29g-MuiContainer-root"
        >
          <div
            class="imageBannerContainer"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt=""
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
